File: W8_SoonZik/SoonZik/Models/Album.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoonZik.Models
{
    public class Album
    {
        public int id { get; set; }
        public string title { get; set; }
        public User user { get; set; }
        public double price { get; set; }
        public int yearProd { get; set; }
        public string image { get; set; }
        public List<Music> musics { get; set; }
    }
}

File: iOS SoonZik/SoonZik/Models/Album.m
//
//  Album.m
//  SoonZik
//
//  Created by LLC on 04/07/2014.
//  Copyright (c) 2014 Coordina. All rights reserved.
//

#import "Album.h"

@implementation Album

- (id)initWithJsonObject:(NSDictionary *)json
{
    self = [super init];
    
    NSLog(@"%@", json);
    
    self.identifier = [[json objectForKey:@"id"] intValue];
    self.image = [json objectForKey:@"image"];
    self.price = [[json objectForKey:@"price"] floatValue];
    self.title = [json objectForKey:@"title"];
    if ([json objectForKey:@"user"]) {
        self.artist = [[User alloc] initWithJsonObject:[json objectForKey:@"user"]];
    }
    self.listOfMusics = [[NSMutableArray alloc] init];
    NSDictionary *musics = [json objectForKey:@"musics"];
    
    if (musics != nil) {
        int i = 0;
        for (NSDictionary *music in musics) {
            Music *m = [[Music alloc] initWithJsonObject:music];
            m.albumImage = self.image;
            [self.listOfMusics addObject:m];
            i++;
        }
    }
    
    return self;
}

@end
